content/browser/child_process_launcher_helper.cc

```diff
@@ -22,6 +24,8 @@
 #include "content/browser/android/launcher_thread.h"
 #endif
 
+int kTcpConnectionTimeout = 3;
```

Comment:
const and how about 5 sec as a default. My hands are too slow.

Reply:
Let me change to 5 sec.

Reply:
When I tried with OnceCallback, Mostly ok but sometimes the logs like below was observed.
Used LOG is,
void NodeController::OnAcceptInvitation() {
  LOG(INFO) << "tcp_success_callback_.is_null(): " << tcp_success_callback_.is_null();
}

I think the reason is some web pages including multiple cross domains uses multiple renderer process for each of these for security reason.
And for this reason, I think RepeatingCallback could be reasonable.

For your reference, I will leave a url with OnceCallback impl.
https://github.com/joonghunpark/Castanets/pull/new/oncecallback

[6117:6154:0911/113107.155153:INFO:child_process_launcher_helper.cc(183)] PostLaunchOnLauncherThread
[6117:6117:0911/113107.155139:INFO:render_process_host_impl.cc(3914)] SKIP!!!!! RenderProcessHostImpl::CreateSharedRendererHistogramAllocator
[6117:6154:0911/113107.155253:INFO:child_process_launcher_helper.cc(213)] PostLaunchOnLauncherThread - mojo::OutgoingInvitation::Send
[6117:6120:0911/113107.156187:INFO:tcp_platform_handle_utils_posix.cc(134)] Server Sock fd for port 37149:498
[6117:6117:0911/113107.165928:INFO:render_process_host_impl.cc(3914)] SKIP!!!!! RenderProcessHostImpl::CreateSharedRendererHistogramAllocator
[6191:6191:0911/113107.618523:WARNING:chrome_main_delegate.cc(280)] process type 'renderer' should be created through the zygote.
[6191:6191:0911/113107.619544:INFO:broker_castanets.cc(147)] Connection Success: Unix Domain Socket
**[6117:6120:0911/113107.619648:INFO:node_controller.cc(971)] tcp_success_callback_.is_null(): 0**
[6192:6192:0911/113107.624037:WARNING:chrome_main_delegate.cc(280)] process type 'renderer' should be created through the zygote.
[6192:6192:0911/113107.625040:INFO:broker_castanets.cc(147)] Connection Success: Unix Domain Socket
**[6117:6120:0911/113107.625115:INFO:node_controller.cc(971)] tcp_success_callback_.is_null(): 1**

Reply:
> When I tried with OnceCallback, Mostly ok but sometimes the logs like below was observed.
> Used LOG is,
> void NodeController::OnAcceptInvitation() {
> LOG(INFO) << "tcp_success_callback_.is_null(): " << tcp_success_callback_.is_null();
> }
> 
> I think the reason is some web pages including multiple cross domains uses multiple renderer process for each of these for security reason.

Seriously? If it is true, Castanets won't work at all. Please share the test site.

> And for this reason, I think RepeatingCallback could be reasonable.

multiple renderer processes -> multiple child process launchers -> multiple invitations -> multiple callbacks BUT that doesn't mean one callback runs multiple times. As I explained, a success callback has to run only once per invitation. Do you agree with this part?

If a success callback is null in AcceptInvitation(), this may be one of the two cases below.
1) you haven't set a proper callback in SendInvitation() -> bug
2) you try to run the same callback for the second time even though the callback had been already run. (The second connection success?) -> bug

> For your reference, I will leave the patch's url with OnceCallback impl.
> https://github.com/joonghunpark/Castanets/commits/oncecallback

Let's just leave this PR with a repeating callback. If necessary, I will create a PR.

Reply:
Seriously? If it is true, Castanets won't work at all. Please share the test site.
> The nullptr OnceCallback occurs only on 2nd Tab. 3rd or 4th is ok it seems.
So I think my assumption about the issue was incorrect. Sorry for my immature assumption.

multiple renderer processes -> multiple child process launchers -> multiple invitations -> multiple callbacks BUT that doesn't mean one callback runs multiple times. As I explained, a success callback has to run only once per invitation. Do you agree with this part?
> I agree with this.

If a success callback is null in AcceptInvitation(), this may be one of the two cases below.

1. you haven't set a proper callback in SendInvitation() -> bug
2. you try to run the same callback for the second time even though the callback had been already run. (The second connection success?) -> bug

> I think this corresponds the case 2.

Let's just leave this PR with a repeating callback. If necessary, I will create a PR.
> I see. Thank you for your review:)